### PR TITLE
fix: use proper option condition in HistoricalStateProvider

### LINKED ISSUE #1951: `eth_getBalance` returns `0` balance instead of `1`
### Describe the bug

Currently the `rpc-compat` hive test for `eth_getBalance/get-balance-blockhash` fails, returning the incorrect balance:
```diff
>>  {"jsonrpc":"2.0","id":1,"method":"eth_getBalance","params":["0xaa00000000000000000000000000000000000000","0x898753d8fdd8d92c1907ca21e68c7970abd290c647a202091181deec3f30a0b2"]}
<<  {"jsonrpc":"2.0","result":"0x0","id":1}
response differs from expected:
 {
   "id": 1,
   "jsonrpc": "2.0",
-  "result": "0x0"
+  "result": "0x1"
 }
```

### Steps to reproduce

Run [hive](#851):
```
./hive --client reth --sim ethereum/rpc-compat --sim.limit "/eth_getBalance"
```

### Node logs

_No response_

### Platform(s)

_No response_

### What version/commit are you on?

_No response_

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by Rjected**
`AccountProvider::basic_account` is used to get basic account information, if it exists. The `HistoricalStateProvider` is able to get this account info for a given `address`, at a given point in the chain history. It does this by checking the `AccountHistory` table for a matching `address`, hoping to obtain a `transition_id`.

If this does not yield a transition id, then there is no account changeset which involves the `address`, and the account info must be obtained from the plain state in `PlainAccountState`.

Previously, we used `let else` semantics to `return Ok(None)` early if we did not obtain a transition id from the above process, which prevented the historical state provider from checking the plain state.

This is an issue when:
 * An account exists at genesis
 * That account is never changed

Instead, we should not return early, allowing us to fetch the account from the plain state.

Fixes #1951

### CODE REVIEW

**Comment by Rjected** [comment]
ah, I see we use `let else` like this in more places in the `HistoricalStateProvider` - adding more fixes

File: crates/storage/provider/src/providers/state/historical.rs [hunk 1]
```diff
@@ -96,26 +102,32 @@ impl<'a, 'b, TX: DbTx<'a>> StateProvider for HistoricalStateProviderRef<'a, 'b,
...
+            .cursor_read::<tables::StorageHistory>()?
             .seek(history_key)?
-            .filter(|(key,_)| key.address == address && key.sharded_key.key == storage_key)
-            .map(|(_,list)| list.0.enable_rank().successor(self.transition as usize).map(|i| i as u64)) else {
-                return Ok(None)
-            };
+            .filter(|(key, _)| key.address == address && key.sharded_key.key == storage_key)
+            .map(|(_, list)| {
+                list.0.enable_rank().successor(self.transition as usize).map(|i| i as u64)
+            });
```
Comment (gakonst) [inline crates/storage/provider/src/providers/state/historical.rs:112]: We should move this to a helper, but OK for now (cc @rakita for when we do the provider refactor)

**Review by gakonst — APPROVED** [review]
ok w me if it fixes the test, cc @mattsse

**Review by mattsse — APPROVED** [review]
nice find

**Pull request merged**